### PR TITLE
GCE network interface whitelist policy

### PR DESCRIPTION
**Pull request by charliewolf**

### CODE REVIEW

File: policies/templates/gcp_compute_network_interface_whitelist_v1.yaml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  name: gcp-compute-network-interface-whitelist-v1
+spec:
+  crd:
+    spec:
+      names:
+        kind: GCPComputeNetworkInterfaceWhitelistConstraintV1
+        plural: gcpcomputenetworkinterfacewhitelistconstraintsV1
+      validation:
+        openAPIV3Schema:
+            whitelist:
```
Comment (AdrienWalkowiak) [inline policies/templates/gcp_compute_network_interface_whitelist_v1.yaml]: same here - could you add a comment for this parameter? (is it mandatory/optional, what is its default value + a valid example for it)

File: policies/templates/gcp_compute_network_interface_whitelist_v1.yaml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            
+            	whitelist := lib.get_default(params, "whitelist", [])
+            	allowed_networks := {n | n = whitelist[_]}
+            
+            	access_configs := lib.get_default(interface, "accessConfig", [])
+            
+            	is_external_network := count(access_configs) > 0
+            	is_network_whitelisted := count({full_network_uri} - allowed_networks) == 0
+            
+            	is_external_network == true
```
Comment (AdrienWalkowiak) [inline policies/templates/gcp_compute_network_interface_whitelist_v1.yaml]: Is there a reason to use a temporary object here? Why not just the test statements directly here (and test for != 0 for L78)?

Comment (charliewolf) [reply policies/templates/gcp_compute_network_interface_whitelist_v1.yaml]: I think creating an `is_external_network` var helps readability.

Comment (AdrienWalkowiak) [reply policies/templates/gcp_compute_network_interface_whitelist_v1.yaml]: fair point :)

**Pull request merged**